Code/PgSQL/rdkit/CMakeLists.txt

```
@@ -1,7 +1,7 @@
 if(APPLE)
   set (EXTENSION_PREFIX "")
-  set (EXTENSION_SUFFIX ".so")
-  set (EXTENSION_DEST_SUFFIX ".so")
+  set (EXTENSION_SUFFIX ".dylib")
+  set (EXTENSION_DEST_SUFFIX ".dylib")
```

I suggest to revert this change as #7869 does the same, but maintaining backwards compatibility with PostgreSQL <16.
```suggestion
  set (EXTENSION_SUFFIX ".so")
  set (EXTENSION_DEST_SUFFIX ".so")
```